component.json

```diff
@@ -5,11 +5,47 @@
   "docsUrl":"https://github.com/elasticio/postgresql-component",
   "credentials": {
     "fields": {
-      "conString": {
-        "label": "Connection String",
+      "protocol": {
+        "viewClass": "TextFieldView",
+        "label": "Protocol",
+        "required": true,
+        "placeholder": "postgresql"
+      },
+      "host": {
+        "viewClass": "TextFieldView",
+        "label": "Connection URI (host)",
         "required": true,
+        "placeholder": "localhost"
+      },
+      "port": {
+        "viewClass": "TextFieldView",
+        "label": "Connection port",
+        "required": false,
+        "placeholder": "1234"
+      },
+      "databaseName": {
+        "viewClass": "TextFieldView",
+        "label": "Database Name",
+        "required": true,
+        "placeholder": "mydb"
+      },
+      "user": {
+        "viewClass": "TextFieldView",
+        "label": "User",
+        "required": true,
+        "placeholder": "User"
+      },
+      "password": {
+        "viewClass": "PasswordFieldView",
+        "label": "Password",
+        "required": true,
+        "placeholder": "Password"
+      },
+      "configurationProperties": {
         "viewClass": "TextFieldView",
-        "placeholder": "postgres://username:password@localhost/database"
+        "label": "Configuration properties",
+        "required": false,
+        "placeholder": "param1=value1&param2&value2"
```

Comment:
I haven't found any mention of an additional connection properties. Seems, it is an extra parameter.